src/Hosting/Hosting/src/StaticWebAssets/StaticWebAssetsLoader.cs

```diff
@@ -21,10 +22,11 @@ public class StaticWebAssetsLoader
         /// <summary>
         /// Configure the <see cref="IWebHostEnvironment"/> to use static web assets.
         /// </summary>
-        /// <param name="environment"></param>
-        public static void UseStaticWebAssets(IWebHostEnvironment environment)
+        /// <param name="environment">The application <see cref="IWebHostEnvironment"/>.</param>
+        /// <param name="configuration">The host <see cref="IConfiguration"/>.</param>
+        public static void UseStaticWebAssets(IWebHostEnvironment environment, IConfiguration configuration)
```

Comment:
I had to add config as a parameter here as the blazor dev-server needs to load the manifest for client-side apps. We will check the host config for a staticwebassets key and use that path if available.

If there is an alternative way of doing this, I'm all ears.

Reply:
@Tratcher FYI

Reply:
This makes sense now you explained it to me offline. For anyone else, the point here is that we don't want to assume it's OK to load the application assembly at runtime just to find out where it is on disk. This applies in the dev server case, but could in other cases too such as automated tests. So it's a good change even besides how it fixes the Blazor dev server case.